homeassistant/components/sensor/hive.py

```diff
@@ -0,0 +1,486 @@
+                if ('value' in snan["later"] and
+                        'start' in snan["later"] and
+                        'Start_DateTime' in snan["later"] and
+                        'End_DateTime' in snan["later"] and
+                        'status' in snan["later"]["value"]):
+                    later_status = snan["later"]["value"]["status"]
+                    later_start = (snan["later"]
+                                   ["Start_DateTime"].strftime("%H:%M"))
+                    later_end = snan["later"]["End_DateTime"].strftime("%H:%M")
+
+                    sa_string = (later_status
+                                 + " : "
+                                 + later_start
+                                 + " - "
+                                 + later_end)
+                    s_a.update({"Later": sa_string})
+        else:
+            s_a.update({"Schedule not active": ""})
+
+        return s_a
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        if self.device_type == "Heating_CurrentTemperature":
+            return TEMP_CELSIUS
+        elif self.device_type == "Heating_TargetTemperature":
+            return TEMP_CELSIUS
+        elif self.device_type == "Heating_State":
+            return None
+        elif self.device_type == "Heating_Mode":
+            return None
+        elif self.device_type == "Heating_Boost":
+            return None
+        elif self.device_type == "HotWater_State":
+            return None
+        elif self.device_type == "HotWater_Mode":
+            return None
+        elif self.device_type == "HotWater_Boost":
+            return None
+        elif self.device_type == "Hive_Device_BatteryLevel":
+            return "%"
+        elif self.device_type == "Hive_Device_Sensor":
+            return None
+        elif self.device_type == "Hive_Device_Light_Mode":
+            return None
+        elif self.device_type == "Hive_Device_Plug_Mode":
+            return None
+
+    @property
+    def icon(self):
+        """Return the icon to use."""
+        device_icon = 'mdi:thermometer'
+
+        if self.device_type == "Heating_CurrentTemperature":
+            device_icon = 'mdi:thermometer'
+        elif self.device_type == "Heating_TargetTemperature":
+            device_icon = 'mdi:thermometer'
+        elif self.device_type == "Heating_State":
+            device_icon = 'mdi:radiator'
+        elif self.device_type == "Heating_Mode":
+            device_icon = 'mdi:radiator'
+        elif self.device_type == "Heating_Boost":
+            device_icon = 'mdi:radiator'
+        elif self.device_type == "HotWater_State":
+            device_icon = 'mdi:water-pump'
+        elif self.device_type == "HotWater_Mode":
+            device_icon = 'mdi:water-pump'
+        elif self.device_type == "HotWater_Boost":
+            device_icon = 'mdi:water-pump'
+        elif self.device_type == "Hive_Device_BatteryLevel":

```

Comment:
Use `icon_for_battery_level` in helpers/icon.py.

Reply:
is this mandatory? what will it do?

Reply:
We should always use the available helpers and utils when we can. See https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/icon.py#L5-L6.
Eg:
```py
device_icon = icon_for_battery_level(battery_level=self.batt_lvl)
```

Reply:
this is good, thanks for this :)
done